components/shared/onboarding/Onboarding.tsx

```diff
@@ -0,0 +1,320 @@
+      email,
+      experiences,
+      newProjects,
+    } = getValues();
+    try {
+      if (nextStep === 2) {
+        await updateUser({
+          updateData: {
+            fullname,
+            portfolio,
+            email,
+          },
+        });
+      }
+      if (nextStep === 3) {
+        await updateUser({
+          updateData: {
+            learningGoals,
+          },
+        });
+      }
+      if (nextStep === 4) {
+        await updateUser({
+          updateData: {
+            experiences: experiences?.map((experience) => experience.name),
+          },
+        });
+      }
+      if (nextStep === 5) {
+        startTransition(() => {
+          updateUser({
+            updateData: {
+              newProjects,
+              startTime: availability?.startTime,
+              endTime: availability?.endTime,
+            },
+          });
+        });
+      }
+    } catch (error) {
+      console.log(error);
+    }
+
+    setCurrentStep(nextStep.toString());
+  };
+  const [selectedFrom, setSelectedFrom] = React.useState<Date>();
+  const [selectedTo, setSelectedTo] = React.useState<Date>();
+
+  const handleDaySelectFrom = (date: Date | undefined) => {
+    if (!date) {
+      setSelectedFrom(date as any);
+      return;
+    }
+
+    const newDate = new Date(
+      date.getFullYear(),
+      date.getMonth(),
+      date.getDate()
+    );
+    setSelectedFrom(newDate as any);
+  };
+  const handleDaySelectTo = (date: Date | undefined) => {
+    if (!date) {
+      setSelectedTo(date as any);
+      return;
+    }
+
+    const newDate = new Date(
+      date.getFullYear(),
+      date.getMonth(),
+      date.getDate()
+    );
+    setSelectedTo(newDate as any);
+  };
+  const onSubmit = async () => {
+    console.log(getValues());
+    try {
+      const {
+        portfolio,
+        availability,
+        learningGoals,
+        email,
+        experiences,
+        newProjects,
+        fullname,
+      } = getValues();
+      const dbExperiences = experiences?.map((experience) => experience.name);
+      const startTime = availability?.startTime;
+      const endTime = availability?.endTime;
+      if (currentStep === "1") {
+        await updateUser({
+          updateData: {
+            portfolio,
+            email,
+            fullname,
+          },
+        });
+      } else if (currentStep === "2") {
+        await updateUser({
+          updateData: {
+            learningGoals,
+          },
+        });
+        changeStep();
+      } else if (currentStep === "3") {
+        await updateUser({
+          updateData: {
+            experiences: dbExperiences,
+          },
+        });
+        changeStep();
+      } else if (currentStep === "4") {
+        await updateUser({
+          updateData: {
+            newProjects,
+            startTime,
+            endTime,
+          },
+        });
+        router.push("/");
+      }
+    } catch (e) {
+      console.log(e);
+    }
```

Comment:
Your onsubmit should never be called when the step is 1 2 or 3. The button shouldn't even be rendered.

This can be your onsubmit:

```js
  const onSubmit = async () => {
    try {
      const { availability, newProjects } = getValues();
      const startTime = availability?.startTime;
      const endTime = availability?.endTime;
      await updateUser({
        updateData: {
          newProjects,
          startTime,
          endTime,
        },
      });
      router.push("/");
    } catch (e) {
      console.log(e);
    }
  };
```

If you want to use the onTransition and pending states here, make sure to make it async and actually 'await' the update user.

```js
      startTransition(async() => {
        await updateUser({
          updateData: {
            newProjects,
            startTime,
            endTime,
          },
        });
      });
```